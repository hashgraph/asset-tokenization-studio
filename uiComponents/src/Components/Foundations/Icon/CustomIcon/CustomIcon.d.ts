import type { ComponentWithAs } from "@chakra-ui/system";
import type { BaseIconProps } from "@/Components/Foundations/Icon";
export interface CustomIconProps extends BaseIconProps {
    name: string;
}
export declare const CustomIcon: ComponentWithAs<"svg", CustomIconProps>;
