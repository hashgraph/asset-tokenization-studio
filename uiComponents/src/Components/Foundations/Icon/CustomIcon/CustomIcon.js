var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { createIcon as createChakraIcon } from "@chakra-ui/icon";
import { forwardRef, useTheme as useChakraTheme } from "@chakra-ui/system";
import _merge from "lodash/merge";
import React from "react";
export var CustomIcon = forwardRef(function (_a, ref) {
    var name = _a.name, props = __rest(_a, ["name"]);
    var configIcons = useChakraTheme().icons;
    if (!(name in configIcons)) {
        throw new Error("Icon '".concat(name, "' not found."));
    }
    var options = _merge(configIcons[name], {
        displayName: name,
        defaultProps: props
    });
    var NewIcon = createChakraIcon(options);
    return React.createElement(NewIcon, { ref: ref });
});
