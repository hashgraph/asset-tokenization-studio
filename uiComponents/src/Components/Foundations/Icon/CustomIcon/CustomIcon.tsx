import type { ComponentWithAs } from "@chakra-ui/system";
import { createIcon as createChakraIcon } from "@chakra-ui/icon";
import { forwardRef, useTheme as useChakraTheme } from "@chakra-ui/system";
import type { ConfigurationIcons } from "@Theme/icons";
import _merge from "lodash/merge";
import React from "react";
import type { BaseIconProps } from "@/Components/Foundations/Icon";

export interface CustomIconProps extends BaseIconProps {
  name: string;
}

export const CustomIcon: ComponentWithAs<"svg", CustomIconProps> = forwardRef<
  CustomIconProps,
  "svg"
>(({ name, ...props }: CustomIconProps, ref) => {
  const { icons: configIcons }: { icons: ConfigurationIcons } =
    useChakraTheme();

  if (!(name in configIcons)) {
    throw new Error(`Icon '${name}' not found.`);
  }

  const options = _merge(configIcons[name], {
    displayName: name,
    defaultProps: props,
  });

  const NewIcon = createChakraIcon(options);

  return <NewIcon ref={ref} />;
});
