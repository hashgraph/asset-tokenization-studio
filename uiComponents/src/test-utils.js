import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";
var DefaultTheme = extendTheme(BasePlatformTheme);
export var render = function (ui, theme) {
    if (theme === void 0) { theme = DefaultTheme; }
    var wrapper = function (_a) {
        var children = _a.children;
        return (React.createElement(ChakraProvider, { theme: theme }, children));
    };
    return testingLibraryRender(ui, { wrapper: wrapper });
};
