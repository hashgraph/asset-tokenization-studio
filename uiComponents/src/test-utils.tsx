import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import type { RenderResult } from "@testing-library/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";

const DefaultTheme = extendTheme(BasePlatformTheme);

export const render = (ui: JSX.Element, theme = DefaultTheme): RenderResult => {
  const wrapper = ({ children }: { children?: React.ReactNode }) => (
    <ChakraProvider theme={theme}>{children}</ChakraProvider>
  );

  return testingLibraryRender(ui, { wrapper });
};
