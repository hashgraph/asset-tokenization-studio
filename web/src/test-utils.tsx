import React from "react";
import { ChakraProvider } from "@chakra-ui/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import type { RenderOptions, RenderResult } from "@testing-library/react";
import { render, waitFor } from "@testing-library/react";
import type { MemoryHistory } from "history";
import { createMemoryHistory } from "history";
import { I18nextProvider } from "react-i18next";
import { Router } from "react-router-dom";
import i18n from "./i18n";
import theme from "./theme";
import userEvent from "@testing-library/user-event";

export const selectCalendar = async (
  component: RenderResult,
  id: string,
  day: string | number = new Date().getDate(),
) => {
  const calendar = component.getByTestId(id);
  await userEvent.click(calendar);

  await waitFor(() => {
    const daysToSelect = component.getAllByTestId(`day-${day}`);

    const dayToSelect = daysToSelect.find(
      (day) => !day.hasAttribute("disabled"),
    );

    if (dayToSelect) {
      userEvent.click(dayToSelect);
    }
  });

  await userEvent.click(document.body);
};

export const testQueryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      refetchOnMount: true,
      cacheTime: 0,
    },
  },
  logger: {
    log: console.log,
    warn: console.warn,
    // âœ… no more errors on the console for tests
    error: () => {},
  },
});

export const queryWrapper = ({ children }: { children: React.ReactNode }) => (
  <QueryClientProvider client={testQueryClient}>{children}</QueryClientProvider>
);

const memoryHistory = createMemoryHistory();

export const AllProviders = ({
  children,
  history = memoryHistory,
}: {
  children?: React.ReactNode;
  history?: MemoryHistory;
}) => {
  return (
    <I18nextProvider i18n={i18n}>
      <QueryClientProvider client={testQueryClient}>
        <ChakraProvider theme={theme}>
          <Router location={history.location} navigator={history}>
            {children}
          </Router>
        </ChakraProvider>
      </QueryClientProvider>
    </I18nextProvider>
  );
};

const customRender = (
  ui: React.ReactElement,
  {
    options,
    history,
  }: {
    options?: RenderOptions;
    history?: MemoryHistory;
  } = {},
): RenderResult =>
  render(ui, {
    wrapper: ({ children }) => (
      <AllProviders history={history}>{children}</AllProviders>
    ),
    ...options,
  });

export { customRender as render };
