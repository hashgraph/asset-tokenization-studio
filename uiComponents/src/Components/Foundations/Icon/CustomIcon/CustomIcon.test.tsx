import { render } from "@/test-utils";
import React from "react";
import type { CustomIconProps } from "./CustomIcon";
import { CustomIcon } from "./CustomIcon";

describe(`<CustomIcon />`, () => {
  const defaultProps = {
    name: "Progress",
  };

  const factoryComponent = (props: CustomIconProps = defaultProps) =>
    render(<CustomIcon {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent();

    expect(component.asFragment()).toMatchSnapshot();
  });

  test("rase exception when icon not found", () => {
    expect(() => factoryComponent({ name: "other" })).toThrowError();
  });
});
