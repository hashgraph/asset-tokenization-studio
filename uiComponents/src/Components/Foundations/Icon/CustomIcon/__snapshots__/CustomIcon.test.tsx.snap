// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomIcon /> renders correctly 1`] = `
<DocumentFragment>
  <svg
    class="chakra-icon css-onkibi"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      clip-rule="evenodd"
      d="M2.04762 12C2.04762 6.50345 6.50345 2.04762 12 2.04762C17.4965 2.04762 21.9524 6.50345 21.9524 12C21.9524 17.4965 17.4965 21.9524 12 21.9524C6.50345 21.9524 2.04762 17.4965 2.04762 12ZM12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1ZM6.71213 7.17424C8.02194 5.73987 9.90561 4.8412 11.9999 4.8412C15.9536 4.8412 19.1587 8.04627 19.1587 11.9999C19.1587 14.138 18.222 16.0566 16.735 17.3691C15.4727 18.4832 13.8158 19.1587 11.9999 19.1587C8.04627 19.1587 4.8412 15.9536 4.8412 11.9999C4.8412 10.1403 5.54958 8.44734 6.71213 7.17424ZM11.9999 3.79358C9.59909 3.79358 7.43846 4.82523 5.93852 6.46781C4.60661 7.92638 3.79358 9.86881 3.79358 11.9999C3.79358 16.5322 7.46769 20.2063 11.9999 20.2063C14.0808 20.2063 15.9819 19.4311 17.4282 18.1546C19.131 16.6516 20.2063 14.451 20.2063 11.9999C20.2063 7.46769 16.5322 3.79358 11.9999 3.79358Z"
      fill="#0F172A"
      fill-rule="evenodd"
      opacity="0.2"
    />
    <path
      clip-rule="evenodd"
      d="M2.04762 12C2.04762 6.50345 6.50345 2.04762 12 2.04762C17.4965 2.04762 21.9524 6.50345 21.9524 12C21.9524 17.4965 17.4965 21.9524 12 21.9524C11.7107 21.9524 11.4762 22.1869 11.4762 22.4762C11.4762 22.7655 11.7107 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 12.2893 1.23452 12.5238 1.52381 12.5238C1.8131 12.5238 2.04762 12.2893 2.04762 12ZM16.7353 17.3692C15.473 18.4833 13.8161 19.1587 12.0002 19.1587C8.04658 19.1587 4.84151 15.9537 4.84151 12C4.84151 9.97288 5.68338 8.14328 7.03798 6.84021C7.24647 6.63965 7.2529 6.30805 7.05234 6.09957C6.85178 5.89108 6.52019 5.88465 6.3117 6.08521C4.7603 7.5776 3.79389 9.67639 3.79389 12C3.79389 16.5323 7.46799 20.2064 12.0002 20.2064C14.0811 20.2064 15.9822 19.4312 17.4285 18.1546C17.6454 17.9632 17.6661 17.6322 17.4746 17.4153C17.2832 17.1984 16.9522 17.1778 16.7353 17.3692Z"
      fill="#0F172A"
      fill-rule="evenodd"
    />
  </svg>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
